stdlib/src/builtin/float_literal.mojo

```diff
@@ -135,7 +135,7 @@ struct FloatLiteral(
         Returns:
             A string representation.
         """
-        return self
```

Comment:
@laszlokindrat can you help me understand how this was working before?  Some weird materialization or implicit conversion rules?  I like being explicit in general, but it's still a bit mysterious how this works both before and after this change.

Reply:
I suspect it would implicitly convert to SIMD which is Stringable and would be implicitly converted through that to `String`. But then I don't really understand how this current change fixes it. Either way @StandinKP could you please make this implementation to just return `Float64(self)` instead?

Reply:
What about [this](https://github.com/modularml/mojo/pull/2563#issuecomment-2099184126)